### DISCUSSION TITLE
Expose all system generated emails as .liquid files

### DISCUSSION THREAD

**Discussion by binaryben** [directus/directus#24872] [Feature Requests]
### Describe the Improvement

The notification email for in-app notifications should be exposed as a `.liquid` file so that users can customise it. If possible, it may be worth seeing if any other system emails are generated that should be customisable and exposed in a template.

**Comment by robinsonjohn** [comment]
From what I can tell, all email templates [reside here](https://github.com/directus/directus/tree/main/api/src/services/mail/templates), which at the moment include:

- `password-reset`
- `user-invitation`

At a minimum, Directus is sending out at least one other type of email, which is the email sent to notify a user of a new in-app notification.

![Screenshot 2024-05-06 at 7 02 16 AM](https://github.com/directus/directus/assets/24327848/0f08f009-fc87-4dc1-9b0f-253d1eaaf430)

**Comment by robinsonjohn** [comment]
In addition, [the documentation](https://docs.directus.io/user-guide/content-module/content/items.html#comment-on-an-item) states that when a person is mentioned by typing the `@` symbol, an email will be sent directing them to the comment.

**Comment by binaryben** [comment]
H/T @connorwinston for pointing out these types of emails just use the `base.liquid` file. Would mentions and emailed notifications be served well by a single new `notification.liquid` template which would probably be blank except for referencing `base.liquid`?

**Comment by robinsonjohn** [comment]
I have customized all three available email template files:

- `base.liquid`
- `password-reset.liquid`
- `user-invitation.liquid`

After doing this, I have noticed the following:

Emails sent alongside in-app notifications **_do not_** use the `base.liquid` template. Also, there is no specific template available for this email.

Emails sent from `@` mentions in a comment **_do use_** the `base.liquid` template, but there is no specific template available for this email either.

So, at a minimum, there are these two additional emails which need a template file to customize. In addition, the `base.liquid` template is being ignored in the in-app notification email.

Also, there currently seems to be no way to customize the email subject for any emails sent by Directus.

**Comment by robinsonjohn** [comment]
Also, the `base.liquid` file makes reference to a footer block, but I don't see any reference to it in the documentation, and again, no .liquid template file available.

```
{% block footer %}{% endblock %}
```

**Comment by robinsonjohn** [comment]
I just found another email sent by Directus which does not have a corresponding liquid template:

When a collection is exported to the file library, an in-app notification is sent to the user when the export is complete. If the user has email notifications turned on, an email is sent letting the user know that the export is ready. 

This email should also have a customizable template.

![Screenshot 2024-05-29 at 9 36 24 AM](https://github.com/directus/directus/assets/24327848/7afe805a-52e3-4d0a-8697-22b421c827a9)

**Comment by Loosie94** [comment]
I am facing the same problem.

It would be nice to be able to adjust the base.liquid template inside of Directus > Appearance. 
At this moment there is hardcoded content inside this mail like: "Sent by the team at" which is not nice when your website is not English.

An extention for this could also do the job.

**Comment by pepeday** [comment]
Is this applicable for Directus Cloud too? Or templates are not accessible there?

**Comment by rijkvanzanten** [comment]
@pepeday Templates can be overridden on the enterprise plan. Those projects run on dedicated hardware meaning that the end user can upload their own trusted files. I don't want to expose that on the self-service cloud tiers yet as the security risk is too great for the feature it unlocks. To René's point above, I like the direction of storing them in the database instead so they're properly user editable.

**Comment by pepeday** [comment]
> @pepeday Templates can be overridden on the enterprise plan. Those projects run on dedicated hardware meaning that the end user can upload their own trusted files. I don't want to expose that on the self-service cloud tiers yet as the security risk is too great for the feature it unlocks. To René's point above, I like the direction of storing them in the database instead so they're properly user editable. 

OK gotcha. From your response I understand that I could possibly use a record from a collection to craft an html email body?
I imagine I'd take a more traditional JS route?

**Comment by rijkvanzanten** [comment]
You can craft new emails through flows with templates from the database, yes! This is purely in regards to overwriting the system emails, eg password-reset or user-invite

**Comment by pepeday** [comment]
> You can craft new emails through flows with templates from the database, yes! This is purely in regards to overwriting the system emails, eg password-reset or user-invite

Ohh OK. Still useful to have but yeah my main usage for now is custom emails, not system generated ones. Thank you!
